### PR TITLE
[PEI][PowerPC] Fix false alarm of stack size limit

### PR DESCRIPTION
**Pull request by bzEq**
PPC64 allows stack size up to ((2^63)-1) bytes. Currently llc reports
```
warning: stack frame size (4294967568) exceeds limit (4294967295) in function 'main'
```
if the stack allocated is larger than 4G.

### CODE REVIEW

File: llvm/test/CodeGen/PowerPC/warn-stack-size.ll [hunk 1]
```diff
@@ -0,0 +1,16 @@
+; RUN: llc -mtriple=powerpc64-unknown-unknown < %s 2>&1 | FileCheck %s
+; RUN: llc -mtriple=powerpc64le-unknown-unknown < %s 2>&1 | FileCheck %s
+
+; CHECK-NOT: warning: {{.*}} stack frame size ({{.*}}) exceeds limit (4294967295) in function 'large_stack'
+define i8* @large_stack() {
+  %s = alloca [281474976710656 x i8], align 1
+  %e = getelementptr i8, i8* %s, i64 0
```
Comment (MaskRay) [inline llvm/test/CodeGen/PowerPC/warn-stack-size.ll:7]: use opaque pointers `ptr`

Does it really work for a larger-than-4G limit? Do you not hit any other limit in LLVM codegen that might assume uint32_t or a smaller limit? You probably want to try more more stack operations.

Comment (bzEq) [reply llvm/test/CodeGen/PowerPC/warn-stack-size.ll]: On PPC64, larger-than-4G stack was supported after https://github.com/llvm/llvm-project/commit/5018a5dcbe70a9e2e39341762ec0ca0ad76d28b4.

Comment (bzEq) [reply llvm/test/CodeGen/PowerPC/warn-stack-size.ll]: And we have some internal users using stack larger than 4G stack.

File: llvm/lib/Target/PowerPC/PPCFrameLowering.cpp [hunk 2]
```diff
@@ -2740,3 +2740,11 @@ bool PPCFrameLowering::enableShrinkWrapping(const MachineFunction &MF) const {
     return false;
   return !MF.getSubtarget<PPCSubtarget>().is32BitELFABI();
 }
+
+uint64_t PPCFrameLowering::getStackThreshold() const {
+  // The scratch register of STUX contains a signed negative 64-bit number.
```
Comment (nickdesaulniers) [inline llvm/lib/Target/PowerPC/PPCFrameLowering.cpp:2745]: I'm not very familiar with ppc, but this comment doesn't mean much to me. Maybe other reviewers with more familiarity can triple check it?

**Review by nickdesaulniers — APPROVED** [review]
LGTM; thanks! :train: :white_check_mark:

**Review by nemanjai — APPROVED** [review]
Were we always getting this warning on the tests added in https://github.com/llvm/llvm-project/commit/5018a5dcbe70a9e2e39341762ec0ca0ad76d28b4?
My initial reaction to this was that the test case needs to show an actual access on a large stack frame, but we already have that in `test/CodeGen/PowerPC/huge-frame-size.ll`
Maybe it would be nice to add these cases to the same file since this is very related.

Otherwise, LGTM

**Comment by bzEq** [comment]
> Were we always getting this warning on the tests added in https://github.com/llvm/llvm-project/commit/5018a5dcbe70a9e2e39341762ec0ca0ad76d28b4?

I think so, I didn't catch that in `FileCheck`. For current llc run on `huge-frame-size.ll`.
```
warning: <unknown>:0:0: stack frame size (4294967344) exceeds limit (4294967295) in function 'foo'
```

**Pull request merged**